enhancements/sig-architecture/30-clusterset-override/README.md

```diff
@@ -0,0 +1,235 @@
+
+#### What kinds of roles exist related to managedClusterSet
+In General, there are three kinds of roles related to managedClusterSet.
+- Cluster admin 
+- Controller/agent which run in managedClusters or hub clusters
+- Normal users
+
+#### What purposes will managedClusterSet may be used for
+a. Create a managedClusterSet for disaster recovery purposes, which should include specific clusters.
+b. Create a managedClusterSet which should include all clusters in one region or cloud provider.
+c. Create a managedClusterSet which has same capacity(like: all clusters enabled middleware)
+d. Create a managedClusterSet for each squad for resource group purposes.
+
+#### Who can add a managedCluster to a managedClusterSet
+In this proposal, managedClusterSet could use any labels to select managedClusters. If one user adds a label to a managedClusters, it may lead to this managedClusters added to a managedClusterSet. So this question also means "who can add/update a label for managedClusters".
+
+1. Cluster admin
+Cluster admin could add any labels to any clusters.
+
+2. Controller/agent which run in managedClusters or hub clusters
+The controller/agent may add labels like: `region: apac`, `cloud: Amazon`, `middlewareEnabled: true`
+These labels should not be changed by the user. And these labels mean these clusters have some attributes.
+- `region: apac` means the cluster is provisioned in apac region
+- `cloud: Amazon` means the cluster is provisioned in Amazon cloud provider
+- `middlewareEnabled: true` menas this cluster enables the middleware.
+
+The managedClusterSet could use these labels to select target clusters.
+
+3. Normal users
+Currently, Any user who has `update` permission to the `managedclusters` could add any labels to the managedCluster. And these labels may lead to the managedClusters added to a managedClusterSet.
+
+But for some managedClusterSet(like: disaster recovery managedClusterSet[a], resource group managedClusterSet[d]), we do not want any users to add their clusters to these managedClusterSets.
+For these managedClusterSets, we hope these managedClusterSets' `spec.clusterLabel.key` should have the same prefix. This prefix may be configured by the user. But in this proposal, we use `cluster.open-cluster-management.io/` as the prefix to identify the managedClusterSet which needs external permissions to add to.
+
+If a user wants to add these kinds of labels, he/she should have an external permission. So we need a way to define this permission.
+Investigate the existing rbac rule in current k8s, we use a rule which is similar to [csr-approver](https://github.com/kubernetes/website/blob/3ca34b9a3be454055ba234f1d2ff7d55809b5040/content/zh/examples/access/certificate-signing-request/clusterrole-approve.yaml#L25)
+We define a subresource `managedclusters/label`, which resource name is the label of managedCluster. This subresource means user have permission to add this label to managedClusters.
```

Comment:
still need to elaborate how to define if a user can add any label keys or values?  Will it require any controller/agent to need this permission also?

Reply:
done